File: tests/nodejs/cases/case_109_serialized_copy_only_01.js
/**
 * Task case 109: serialized_copy_only_01 false-positive/false-negative stress pattern.
 */

const retainedCache = new Map();
const retainedAudit = [];
const retainedHandlers = [];

function case109SerializedCopyOnly01(input) {
  const taskName = 'serialized_copy_only_01';
  const raw = input || 'sample';
  const payload = {
    task: taskName,
    entity: 'stress',
    stage: 'evaluation',
    input: raw,
    checksum: `${taskName}:${raw.length}`,
  };
  const serialized = JSON.stringify(payload);
  // SAFE: only primitive serialized data leaves scope.
  return serialized;
}

module.exports = {
  case109SerializedCopyOnly01,
};

File: tests/go/cases/case_109_serialized_copy_only_01.go
package escape_tests

var retainedCase109 = []map[string]string{}

func Case109SerializedCopyOnly01(input string) string {
	raw := input
	if raw == "" {
		raw = "sample"
	}
	payload := map[string]string{
		"task": "serialized_copy_only_01",
		"entity": "stress",
		"stage": "evaluation",
		"input": raw,
		"checksum": "serialized_copy_only_01:" + raw,
	}
	serialized := payload["task"] + ":" + payload["input"]
	// SAFE: only primitive string leaves function.
	return serialized
}
